cmd/ethrex/launch.rs

```diff
@@ -0,0 +1,605 @@
+    sync::SyncManager,
+    types::{Node, NodeRecord},
+};
+use ethrex_storage::{EngineType, Store};
+use ethrex_vm::{backends::EVM, EVM_BACKEND};
+use k256::ecdsa::SigningKey;
+use local_ip_address::local_ip;
+use rand::rngs::OsRng;
+use tokio_util::task::TaskTracker;
+use tracing::{error, info, warn};
+use tracing_subscriber::{filter::Directive, EnvFilter, FmtSubscriber};
+
+const DEFAULT_DATADIR: &str = "ethrex";
+
+#[cfg(feature = "l2")]
+pub async fn launch_l2(matches: clap::ArgMatches) {
+    if let Some(matches) = matches.subcommand_matches("removedb") {
+        let data_dir = matches
+            .get_one::<String>("datadir")
+            .map_or(set_datadir(DEFAULT_DATADIR), |datadir| set_datadir(datadir));
+        let path = Path::new(&data_dir);
+        if path.exists() {
+            std::fs::remove_dir_all(path).expect("Failed to remove data directory");
+        } else {
+            warn!("Data directory does not exist: {}", data_dir);
+        }
+        return;
+    }
+
+    let log_level = matches
+        .get_one::<String>("log.level")
+        .expect("shouldn't happen, log.level is used with a default value");
+    let log_filter = EnvFilter::builder()
+        .with_default_directive(
+            Directive::from_str(log_level).expect("Not supported log level provided"),
+        )
+        .from_env_lossy();
+    let subscriber = FmtSubscriber::builder()
+        .with_env_filter(log_filter)
+        .finish();
+    tracing::subscriber::set_global_default(subscriber).expect("setting default subscriber failed");
+
+    let http_addr = matches
+        .get_one::<String>("http.addr")
+        .expect("http.addr is required");
+    let http_port = matches
+        .get_one::<String>("http.port")
+        .expect("http.port is required");
+    let authrpc_addr = matches
+        .get_one::<String>("authrpc.addr")
+        .expect("authrpc.addr is required");
+    let authrpc_port = matches
+        .get_one::<String>("authrpc.port")
+        .expect("authrpc.port is required");
+    let authrpc_jwtsecret = matches
+        .get_one::<String>("authrpc.jwtsecret")
+        .expect("authrpc.jwtsecret is required");
+
+    let tcp_addr = matches
+        .get_one::<String>("p2p.addr")
+        .expect("addr is required");
+    let tcp_port = matches
+        .get_one::<String>("p2p.port")
+        .expect("port is required");
+    let udp_addr = matches
+        .get_one::<String>("discovery.addr")
+        .expect("discovery.addr is required");
+    let udp_port = matches
+        .get_one::<String>("discovery.port")
+        .expect("discovery.port is required");
+
+    let mut network = matches
+        .get_one::<String>("network")
+        .expect("network is required")
+        .clone();
+
+    let mut bootnodes: Vec<Node> = matches
+        .get_many("bootnodes")
+        .map(Iterator::copied)
+        .map(Iterator::collect)
+        .unwrap_or_default();
+
+    if network == "holesky" {
```

Comment:
These networks specific checks can be removed for L2. In general, a whole bunch of other initialization stuff related to p2p can be removed, though it might be a bit of work so it's fine to leave for another PR